crates/common/src/clock.rs

```diff
@@ -298,19 +298,25 @@ impl Clock for TestClock {
         callback: Option<TimeEventCallback>,
     ) -> anyhow::Result<()> {
         check_valid_string(name, stringify!(name))?;
+        let name_ustr = Ustr::from(name);
+
         check_predicate_true(
-            callback.is_some() | self.default_callback.is_some(),
+            callback.is_some()
+                | self.callbacks.contains_key(&name_ustr)
+                | self.default_callback.is_some(),
             "No callbacks provided",
         )?;
 
-        let name_ustr = Ustr::from(name);
         match callback {
             Some(callback_py) => self.callbacks.insert(name_ustr, callback_py),
             None => None,
         };
 
-        let ts_now = self.time.get_time_ns();
+        // This allows to reuse a time alert without updating the callback, for example for non regular monthly alerts.
```

Comment:
We could also push this logic [here](https://github.com/nautechsystems/nautilus_trader/blob/develop/nautilus_trader/common/component.pyx#L337) down to Rust, as I think it is achieving the same thing.
(can be in a follow-up PR.)